### PR TITLE
[Master] Remove `on fail` from TransactionalNode and RetryStatementNode

### LINKED ISSUE #37315: On fail should only be allowed with regular-compound-stmt
**Description:**
Ref https://ballerina.io/spec/lang/master/#stmt-with-on-fail

**Steps to reproduce:**

**Affected Versions:**
2201.1.1

**OS, DB, other environment details and versions:**

**Related Issues (optional):**

**Suggested Labels (optional):**

**Suggested Assignees (optional):**

### PR DESCRIPTION
**Pull request by SasinduDilshara**
Fixes #37315

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java [hunk 1]
```diff
@@ -13633,8 +13633,7 @@ private STNode parseTransactionStatement(STNode transactionKeyword) {
         startContext(ParserRuleContext.TRANSACTION_STMT);
         STNode blockStmt = parseBlockNode();
         endContext();
-        STNode onFailClause = parseOptionalOnFailClause();
-        return STNodeFactory.createTransactionStatementNode(transactionKeyword, blockStmt, onFailClause);
+        return STNodeFactory.createTransactionStatementNode(transactionKeyword, blockStmt);
```
Comment (lochana-chathura) [inline compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java:13636]: We can remove the optional `onFailClause` in the ST also right?
https://github.com/ballerina-platform/ballerina-lang/blob/2d730a699ebe7ff6be930cfe7e47a2b09c38473c/compiler/ballerina-treegen/src/main/resources/syntax_tree_descriptor.json#L3595-L3599

Comment (lochana-chathura) [reply compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java:13636]: Seems the changes to `syntax_tree_descriptor.json` is not committed. Also when we change it we have to update `api_gen_syntax_tree_descriptor.json` as well.

File: compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java [hunk 2]
```diff
@@ -13717,8 +13716,7 @@ private STNode parseRetryTypeParamRhs(STNode retryKeyword, STNode typeParam) {
 
         STNode blockStmt = parseRetryBody();
         endContext(); // end retry-stmt
-        STNode onFailClause = parseOptionalOnFailClause();
-        return STNodeFactory.createRetryStatementNode(retryKeyword, typeParam, args, blockStmt, onFailClause);
+        return STNodeFactory.createRetryStatementNode(retryKeyword, typeParam, args, blockStmt);
```
Comment (lochana-chathura) [inline compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java:13719]: Let's remove the error handler path added for on-fail-clause these changes. 

https://github.com/ballerina-platform/ballerina-lang/blob/2d730a699ebe7ff6be930cfe7e47a2b09c38473c/compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParserErrorHandler.java#L4735

Comment (SasinduDilshara) [reply compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java:13719]: I will remove it.

File: compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java (hunk 1 above)
Comment (SasinduDilshara) [reply compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java:13636]: Updated. 6e28d27522656155f4d8712f60a34356460eecca

File: compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java (hunk 2 above)
Comment (SasinduDilshara) [reply compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java:13719]: Updated. 6e28d27522656155f4d8712f60a34356460eecca

**Review by KavinduZoysa — APPROVED** [review]

**Review by pcnfernando — CHANGES_REQUESTED** [review]
Let's hold this for a little

**Pull request closed without merging**